_data-prepper/peer-forwarder.md

```diff
@@ -0,0 +1,183 @@
+
+Your Data Prepper configuration needs to include:
+* `aws_cloud_map_namespace_name` - Set to your Cloud Map Namespace name
+* `aws_cloud_map_service_name` - Set to the service name within your specified Namespace
+* `aws_region` - The AWS region where your namespace exists.
+* `discovery_mode` - Set to `aws_cloud_map`
+
+Your Data Prepper configuration can optionally include:
+* `aws_cloud_map_query_parameters` - Key/value pairs to filter the results based on the custom attributes attached to an instance. Only instances that match all the specified key-value pairs are returned.
+
+### Example configuration
+
+```yaml
+peer_forwarder:
+  discovery_mode: aws_cloud_map
+  aws_cloud_map_namespace_name: "my-namespace"
+  aws_cloud_map_service_name: "data-prepper-cluster"
+  aws_cloud_map_query_parameters:
+    instance_type: "r5.xlarge"
+  aws_region: "us-east-1"
+```
+
+### IAM policy with necessary permissions
+
+The Data Prepper must also be running with the necessary permissions. The following IAM policy shows the necessary permissions.
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "CloudMapPeerForwarder",
+            "Effect": "Allow",
+            "Action": "servicediscovery:DiscoverInstances",
+            "Resource": "*"
+        }
+    ]
+}
+```
+---
+## Configuration
+
+See the following for optional configuration values.
+
+### Optional configuration
+
+| Value | Description |
+| ----  | ----------- |
+| `port` | An `int` between 0 and 65535 represents the port peer forwarder server is running on. Default value is `4994`.|
+| `request_timeout` | Duration - An `int` representing the request timeout in milliseconds for Peer Forwarder HTTP server. Default value is `10000`. |
+| `server_thread_count` | An `int` representing number of threads used by Peer Forwarder server. Defaults to `200`.|
+| `client_thread_count` | An `int` representing number of threads used by Peer Forwarder client. Defaults to `200`.|
+| `maxConnectionCount`  |  An `int` representing maximum number of open connections for Peer Forwarder server. Default value is `500`. |
+| `discovery_mode` | A `String` representing the peer discovery mode to be used. Allowable values are `local_node`, `static`, `dns`, and `aws_cloud_map`. Defaults to `local_node` which processes events locally.  |
+| `static_endpoints` |  A `list` containing endpoints of all Data Prepper instances. Required if `discovery_mode` is set to `static`. |
+|  `domain_name` | A `String` representing single domain name to query DNS against. Typically, used by creating multiple [DNS A Records](https://www.cloudflare.com/learning/dns/dns-records/dns-a-record/) for the same domain. Required if `discovery_mode` is set to `dns`.  |
+| `aws_cloud_map_namespace_name`  | A `String` representing the Cloud Map namespace when using AWS Cloud Map service discovery. Required if `discovery_mode` is set to `aws_cloud_map`.  |
```

Comment:
Should "namespace" be capitalized? It's treated as a proper name in preceding sections.